core/src/main/scala/org/apache/spark/scheduler/ShuffleMapTask.scala

```diff
@@ -92,6 +92,12 @@ private[spark] class ShuffleMapTask(
       threadMXBean.getCurrentThreadCpuTime - deserializeStartCpuTime
     } else 0L
 
+    // Register the shuffle write metrics reporter to shuffleWriteMetrics.
+    if (dep.shuffleWriteMetricsReporter.isDefined) {
+      context.taskMetrics().shuffleWriteMetrics.registerExternalShuffleWriteReporter(
```

Comment:
This happens per-task, I think `ShuffleWriteMetrics.externalReporters` can be option instead of array.